network-config-analyzer/IstioPolicyYamlParser.py

```diff
@@ -290,21 +291,68 @@ def parse_regex_dimension_values(self, dim_name, values_list, negative_values_li
         return res
 
     @staticmethod
-    def _get_connection_set_from_properties(dest_ports, methods_dfa=None, paths_dfa=None, hosts_dfa=None):
+    def _get_connection_set_from_properties(dest_ports, method_set=MethodSet(True), paths_dfa=None, hosts_dfa=None):
         """
         get ConnectionSet with TCP allowed connections, corresponding to input properties cube
         :param PortSet dest_ports: ports set for dset_ports dimension
-        :param MinDFA methods_dfa: MinDFA obj for methods dimension
+        :param MethodSet method_set: methods set for methods dimension
         :param MinDFA paths_dfa: MinDFA obj for paths dimension
         :param MinDFA hosts_dfa: MinDFA obj for hosts dimension
         :return: ConnectionSet with TCP allowed connections , corresponding to input properties cube
         """
-        tcp_properties = TcpLikeProperties(source_ports=PortSet(True), dest_ports=dest_ports, methods=methods_dfa,
+        tcp_properties = TcpLikeProperties(source_ports=PortSet(True), dest_ports=dest_ports, methods=method_set,
                                            paths=paths_dfa, hosts=hosts_dfa)
         res = ConnectionSet()
         res.add_connections('TCP', tcp_properties)
         return res
 
+    def _parse_method(self, method_str, methods_array):
+        res = MethodSet()
+        if method_str == '*': # presence match
+            res.add_interval(MethodSet.whole_range_interval())
+            return res
+        if '*' in method_str:
```

Comment:
Maybe we need a more general function for validity of str values in istio policies, for all supported fields.
The condition on where `*` can appear are general and not only for methods.
In our supported fields this validation is also relevant for hosts, paths attributes.

Reply:
Done